src/packet/signature/ser.rs

```diff
@@ -286,7 +315,16 @@ impl SignatureConfig {
             packet.to_writer(&mut unhashed_subpackets)?;
         }
 
-        writer.write_u16::<BigEndian>(unhashed_subpackets.len() as u16)?;
+        match self.version {
+            SignatureVersion::V4 => {
+                writer.write_u16::<BigEndian>(unhashed_subpackets.len() as u16)?
```

Comment:
should use `try_from` for these

Reply:
(footnote: I've caught a few more "as" casts, in the meantime, and fixed them in https://github.com/hko-s/rpgp/pull/2)